clients/python-client/python_client/utils/kuberay_cluster_utils.py

```diff
@@ -0,0 +1,487 @@
+
+        Returns:
+        - Tuple[dict, bool]: A tuple of the cluster specification and a boolean indicating
+            whether the worker group was successfully populated.
+        """
+        # validate arguments
+        try:
+            arguments = locals()
+            for k, v in arguments.items():
+                if k != "min_replicas":
+                    assert v
+        except AssertionError as e:
+            log.error(
+                "error populating worker group, the parameters are not fully defined. {} = {}".format(
+                    k, v
+                )
+            )
+            return cluster, False
+
+        assert self.is_valid_name(group_name)
+        assert max_replicas >= min_replicas
+
+        worker_group: dict[str, Any] = {
+            "groupName": group_name,
+            "maxReplicas": max_replicas,
+            "minReplicas": min_replicas,
+            "rayStartParams": ray_start_params,
+            "replicas": replicas,
+            "template": {
+                "spec": {
+                    "containers": [
+                        {
+                            "image": ray_image,
+                            "command": ray_command,
+                            "lifecycle": {
+                                "preStop": {
+                                    "exec": {"command": ["/bin/sh", "-c", "ray stop"]}
+                                }
+                            },
+                            "name": "ray-worker",
+                            "resources": {
+                                "requests": {
+                                    "cpu": cpu_requests,
+                                    "memory": memory_requests,
+                                },
+                                "limits": {
+                                    "cpu": cpu_limits,
+                                    "memory": memory_limits,
+                                },
+                            },
+                            "volumeMounts": [
+                                {"mountPath": "/tmp/ray", "name": "ray-logs"}
+                            ],
+                        }
+                    ],
+                    "initContainers": [
+                        {
+                            "command": [
+                                "sh",
+                                "-c",
+                                "until nslookup $RAY_IP.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for K8s Service $RAY_IP; sleep 2; done",
+                            ],
+                            "image": init_image,
+                            "name": "init",
+                        }
+                    ],
+                    "volumes": [{"emptyDir": {}, "name": "ray-logs"}],
+                }
+            },
+        }
+
+        return worker_group, True
```

Comment:
Wondering if there is a particular reason to return only the `worker_group ` instead of  `cluster`. Because it seems to me all other similar functions(`populate_ray_head`, `populate_meta`) return the whole `cluster`.

Reply:
good point. this is mainly because multiple actors can be creating worker-groups for the same cluster, and eventually we would merge them into one cluster.

at a high level, something lie this:

actor1 --> populate(worker_group1)
actor2 --> populate(worker_group2)
cluster.workgroupspecs.append(worker_group1, worker_group2)

Reply:
I see your point. Thank you!